applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -283,7 +283,8 @@ def FinalizeSolutionStep(self):
             self._GetSolutionStrategy().FinalizeSolutionStep()
 
             # Limit the obtained acceleration for the next step
-            if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] > 3:
+            # This limitation should be called on the second solution step onwards (e.g. STEP=3 for BDF2).
+            if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= self.min_buffer_size:
```

Comment:
```suggestion
            # This limitation should be called on the second solution step onwards (e.g. STEP=3 for BDF2)
            # We intentionally avoid correcting the acceleration in the first resolution step as this might cause problems with zero initial conditions
            if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= self.min_buffer_size:
```